src/modules/cpu/kernel/non_silent_region_detection.hpp

```diff
@@ -20,6 +20,60 @@ OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 THE SOFTWARE.
 */
 
+
+/* Non Silent Region Detection requires Moving Mean Square (MMS) computation on input audio data
+MMS buffer is a 1D buffer having same length as input audio. The algorithm used for MMS computation is explained with a sample use case
+
+Example:
+Input: [1, 2, 3, 4, 5, 6, 7, 8]
+audio_length = 8
+window_length = 3
+reset_interval_length = 4
+
+window_begin = -window_length + 1 = -2
+window_factor = 1 / window_length = 1/3
+
+MMS computation is divided into blocks of reset interval length
+num_blocks = audio_length / reset_interval_length
+For the above example we will have
+    - 2 blocks (8 / 4)
+    - each block runs for 4 iterations
+    - in each iteration window begin value is increment by 1
+
+Block1
+window begin = -2
+Iteration 0:    sum_of_squares = 1*1                              // window begin = -2
+                store sum_of_squares * window_factor in MMS[0]
+
+Iteration 1:    sum_of_squares = 1*1 + 2*2                        // window begin = -1
+                store sum_of_squares * window_factor in MMS[1]
+
+Iteration 2:    sum_of_squares = 1*1 + 2*2 + 3*3                  // window begin =  0
+                store sum_of_squares * window_factor in MMS[2]
+                sum_of_squares -= 1*1
+
+Iteration 3:    sum_of_squares = 2*2 + 3*3 + 4*4                  // window begin =  1
+                store sum_of_squares * window_factor in MMS[3]
+                sum_of_squares -= 2*2
+
+Block2
+Iteration 0:    sum_of_squares = 3*3 + 4*4 + 5*5                  // window begin = 2
+                store sum_of_squares * window_factor in MMS[4]
+                sum_of_squares -= 3*3
+
+Iteration 1:    sum_of_squares = 4*4 + 5*5 + 6*6                 // window begin = 3
+                store sum_of_squares * window_factor in MMS[5]
+                sum_of_squares -= 4*4
+
+Iteration 2:    sum_of_squares = 5*5 + 6*6 + 7*7                 // window begin = 4
+                store sum_of_squares * window_factor in MMS[6]
+                sum_of_squares -= 5*5
+
+Iteration 3:    sum_of_squares  = 6*6 + 7*7 + 8*8                // window begin = 5
+                store sum_of_squares * window_factor in MMS[7]
+                sum_of_squares -= 6*6
```

Comment:
Example looks good, but from someone looking at it the first time, they'll still expect what we do after this, to find the non-silent region's index and length. Either finish the example, or state in words.